sklearn/tree/tests/test_tree.py

```diff
@@ -994,6 +997,24 @@ def test_min_impurity_decrease():
         serialized_object = pickle.dumps(est)
         est2 = pickle.loads(serialized_object)
         assert type(est2) == est.__class__
+
+        # test that all class properties are maintained
+        for prop_name in [
```

Comment:
We can extend the existing `fitted_attributes` part to also test these parameters:

```python
        attributes = [
            "max_depth",
            "node_count",
            "capacity",
            "n_classes",
            "children_left",
            "children_right",
            "n_leaves",
            "feature",
            "threshold",
            "impurity",
            "n_node_samples",
            "weighted_n_node_samples",
            "value",
        ]
        fitted_attribute = {
            attribute: getattr(est.tree_, attribute) for attribute in attributes
        }
```

and then alter the test below to be able to check for scalars and arrays:

```python
        for attribute in fitted_attribute:
            assert_array_equal(
                getattr(est2.tree_, attribute),
                fitted_attribute[attribute],
                err_msg=(
                    f"Failed to generate same attribute {attribute} after pickling with"
                    f" {name}"
                ),
            )
```

Reply:
Done.